cli/sql.go

```diff
@@ -74,11 +75,47 @@ http://www.cockroachlabs.com/docs/
 `)
 }
 
+// handleSet handles the \set command and changes client-side flags.
+func handleSet(cmd []string, confirm *bool) int {
+	if len(cmd) != 2 {
+		fmt.Fprintf(os.Stderr, "Usage:\n \\set <option>\n")
+		return cliNextLine
+	}
+
+	switch cmd[1] {
+	case `confirm`:
+		*confirm = true
+	case `noconfirm`:
+		*confirm = false
+	default:
+		fmt.Fprintf(os.Stderr, "unrecognized \\set option: %s\n", cmd[1])
+	}
+	return cliNextLine
+}
+
+// confirmStatement checks whether the statement string contains
+// "dangerous" statements and asks a confirmation to the user if
+// necessary.
+func confirmStatement(confirm bool, stmt string) bool {
+	stmt = strings.ToUpper(stmt)
+	if confirm && (strings.Contains(stmt, "DROP ")) {
```

Comment:
Matching `/drop /i` anywhere in the query string seems overly broad. If we're not going to use a real query parser we should only look at the first word of the query (even though this will miss use of `DROP` within `ALTER TABLE` statements)